internal/egress/egress.go

```diff
@@ -1,65 +1,133 @@
 package egress
 
 import (
+	"context"
 	"errors"
+	"github.com/open-bastion/open-bastion/internal/datastore"
+	"github.com/open-bastion/open-bastion/internal/obclient"
 	"io"
 	"strconv"
-	"strings"
+	"time"
 
 	"golang.org/x/crypto/ssh"
 
-	logger "github.com/open-bastion/open-bastion/internal/logger"
+	"github.com/open-bastion/open-bastion/internal/logger"
 )
 
-// BackendConn contains the informations to establish a connection to a backend
-type BackendConn struct {
-	Command string
-	User    string
-	Host    string
-	Port    int
+//EstablishSSHConnection takes a client connected with SSH to the bastion and tries to get its information from the
+//datastore to establish a connection to the backend.
+func EstablishSSHConnection(ctx context.Context, client *obclient.Client, dataStore datastore.DataStore) {
+	var err error
+	//The user has already been validated during the ssh handshake and should be good
+	//We use the connecting user to parse its key
+	client.SSHKey, err = dataStore.GetUserEgressPrivateKeySigner(client.User)
+
+	logger.UpdateClientLogCtx(ctx, client)
+
+	if err != nil {
+		_, _ = client.SshCommChan.Write([]byte("error accessing credentials"))
+
+		logger.ErrorWithCtxWithErr(ctx, err, "authenticated user could not access his egress private key")
+	}
+
+	//TODO add proper cancel func to clean up and display an error
+	ctx, cancel := context.WithTimeout(ctx, time.Millisecond*1000)
```

Comment:
1 second is way too low, we might want to increase it and even make this a parameter

Reply:
Will be done in a future PR!